### PR TITLE
PackageItem schema test

### PR DESCRIPTION
**Pull request by Jeshan7**
## Why?
  - Coverage was low

## This change addresses the need by:
     - Updated the test cases in schema/package_item_test.exs
     - Wrote new test cases in schema/package_item_test.exs
        - Test cases for foreign_keys(product_id, packege_id, line_item_id)
        - Test cases for update_changeset/2

    [delivers #163248592]

## Checklist

- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

### CODE REVIEW

File: apps/snitch_core/test/data/schema/package_item_test.exs [hunk 1]
```diff
@@ -19,8 +20,12 @@ defmodule Snitch.Data.Schema.PackageItemTest do
     shipping_tax: nil
   }
 
+  setup :zones
+  setup :shipping_methods
+  setup :embedded_shipping_methods
+
   describe "create_changeset/2" do
-    test "with valid params, and backorder is computed correctly" do
+    test "suuceesfully with valid params, and backorder is computed correctly" do
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/schema/package_item_test.exs]: Typo `suuceesfully`

File: apps/snitch_core/test/data/schema/package_item_test.exs [hunk 2]
```diff
@@ -91,5 +123,79 @@ defmodule Snitch.Data.Schema.PackageItemTest do
...
+      cs = PackageItem.update_changeset(%PackageItem{}, Map.from_struct(package_item))
+
+      assert %{
+               shipping_tax: ["must be equal or greater than 0"],
+               tax: ["must be equal or greater than 0"]
+             } = errors_on(cs)
+    end
+  end
+
+  defp new_package(context) do
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/schema/package_item_test.exs]: This method setup the context with things related to package.

Can we rename this function with proper name?
**OR**
We can setup thing individually in a particular order as follows:
```elixir
setup :product
setup :line_item
setup :package
```
@arjun289 Any thoughts on this?

File: apps/snitch_core/test/data/schema/package_item_test.exs [hunk 3]
```diff
@@ -91,5 +123,79 @@ defmodule Snitch.Data.Schema.PackageItemTest do
...
+    end
+
+    test "fails with invalid and valid tax, shipping tax", context do
+      bad_money = Money.new(-1, :USD)
+      %{package_item: package_item} = make_package_item(context)
+      cs = PackageItem.create_changeset(%PackageItem{}, Map.from_struct(package_item))
+      refute cs.valid?
+
+      package_item = %{package_item | tax: bad_money, shipping_tax: bad_money}
+      cs = PackageItem.update_changeset(%PackageItem{}, Map.from_struct(package_item))
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/schema/package_item_test.exs]: We should use already inserted Package item for update.

File: apps/snitch_core/test/data/schema/package_item_test.exs [hunk 4]
```diff
@@ -91,5 +123,79 @@ defmodule Snitch.Data.Schema.PackageItemTest do
...
       assert {:ok, false} = fetch_change(cs, :backordered?)
     end
+
+    test "fails with invalid quantity, delta", context do
+      %{package_item: package_item} = make_package_item(context)
+      cs = PackageItem.create_changeset(%PackageItem{}, Map.from_struct(package_item))
+      refute cs.valid?
+
+      package_item = %{package_item | quantity: -3, delta: -2}
+      cs = PackageItem.update_changeset(%PackageItem{}, Map.from_struct(package_item))
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/schema/package_item_test.exs]: Use already inserted package item for update

**Review by SagarKarwande — CHANGES_REQUESTED** [review]
Format the files with Elixir formatter

File: apps/snitch_core/test/data/schema/package_item_test.exs [hunk 5]
```diff
@@ -41,7 +46,34 @@ defmodule Snitch.Data.Schema.PackageItemTest do
              } == errors_on(cs)
     end
 
-    test "with invalid quantity, delta" do
+    test "fails for non-existent line_item_id", context do
```
Comment (arjun289) [inline apps/snitch_core/test/data/schema/package_item_test.exs]: Group together the tests for missing foreign keys in one test.

Comment (Jeshan7) [reply apps/snitch_core/test/data/schema/package_item_test.exs]: By grouping them together it checks only for one foreign key constraint and ignores the other

File: apps/snitch_core/test/data/schema/package_item_test.exs (hunk 2 above)
Comment (arjun289) [reply apps/snitch_core/test/data/schema/package_item_test.exs]: Yes it can be done this way

**Review by SagarKarwande — APPROVED** [review]

**Review by arjun289 — APPROVED** [review]

**Pull request merged**